tests/manage/z_cluster/test_must_gather.py

```diff
@@ -1,270 +1,49 @@
-                must_gather_helper.match(logs)
-            )
-        ]
-        logger.info(f"Logs: {logs}")
-        logger.info(f"pods list: {pods}")
-        assert set(sorted(logs)) == set(sorted(pods)), (
-            f"List of openshift-storage pods are not equal to list of logs "
-            f"directories list of pods: {pods} list of log directories: {logs}"
-        )
-
-        # 2nd test: Verify logs file are not empty
-        logs_dir_list = self.search_log_files(directory)
-        assert self.check_file_size(logs_dir_list), (
-            "One or more log file are empty"
-        )
-
-        # Find must_gather_commands directory for verification
-        for dir_root, dirs, files in os.walk(directory + "_ocs_logs"):
-            if os.path.basename(dir_root) == 'must_gather_commands':
-                logger.info(
-                    f"Found must_gather_commands directory - {dir_root}"
-                )
-                assert 'json_output' in dirs, (
-                    "json_output directory is not present in "
-                    "must_gather_commands directory."
-                )
-                assert files, (
-                    "No files present in must_gather_commands directory."
-                )
-                cmd_files_path = [
-                    os.path.join(dir_root, file_name) for file_name in files
-                ]
-                json_output_dir = os.path.join(dir_root, 'json_output')
-                break
-
-        # Verify that command output files are present as expected
-        assert set(constants.MUST_GATHER_COMMANDS).issubset(files), (
-            f"Actual and expected commands output files are not matching.\n"
-            f"Actual: {files}\nExpected: {constants.MUST_GATHER_COMMANDS}"
-        )
-        if sorted(constants.MUST_GATHER_COMMANDS_JSON) != sorted(files):
-            logger.warning(
-                "There are more actual must gather commands than expected"
-            )
-
-        # Verify that files for command output in json are present as expected
-        commands_json = os.listdir(json_output_dir)
-        assert set(constants.MUST_GATHER_COMMANDS_JSON).issubset(commands_json), (
-            f"Actual and expected json output commands files are not "
-            f"matching.\nActual: {commands_json}\n"
-            f"Expected: {constants.MUST_GATHER_COMMANDS_JSON}"
-        )
-        if sorted(constants.MUST_GATHER_COMMANDS_JSON) != sorted(commands_json):
-            logger.warning(
-                "There are more actual must gather commands than expected"
-            )
-
-        # Verify that command output files are not empty
-        empty_files = []
-        json_cmd_files_path = [
-            os.path.join(json_output_dir, file_name) for file_name in commands_json
+    @tier1
+    @pytest.mark.parametrize(
+        argnames=['log_type'],
+        argvalues=[
+            pytest.param(
+                *['CEPH'], marks=pytest.mark.polarion_id("OCS-XXX1")
```

Comment:
IIUC this TC is parametrized and it will run 3 times whole must gather right?

Isn't better approach to have class scope fixture which will run must gather only once to save time? And then have those data just verified in 3 separated test cases which will have the same data?
Or it can be defined as parametrized but try to collect must gather only once to save time.

Reply:
@petr-balogh Done, I used `scope='session'`